python/strings_udf/strings_udf/__init__.py

```diff
@@ -59,13 +64,33 @@ def compiler_from_ptx_file(path):
                 "files. Please file an issue reporting this error and how you "
                 "installed cudf and strings_udf."
             )
-        sms = [
-            os.path.basename(f).rstrip(".ptx").lstrip("shim_") for f in files
-        ]
-        selected_sm = max(sm for sm in sms if sm < cc)
-        ptxpath = os.path.join(
-            os.path.dirname(__file__), f"shim_{selected_sm}.ptx"
-        )
+
+        virtual_sms = []
+        native_sms = []
+        for f in files:
+            file_name = os.path.basename(f)
+            sm_number = int(
```

Comment:
this can't be a number. SM's aren't all numeric values any more

Reply:
To be clear the `dev.compute_capability` value will always be an integer. ( 75, 80, 90, ... ). But going forward we will see code gen with values like 90a which means they hold sm90 only instructions that can't be used by newer archs